mungegithub/mungers/mungerutil/util.go

```diff
@@ -16,14 +16,137 @@ limitations under the License.
+
+	// Fill-in map of all the users in each lists (remove duplicates)
+	for _, userslist := range userslists {
+		for _, user := range userslist {
+			uniqueUsers[user] = struct{}{}
+		}
+	}
+
+	// Get the list of unique users
+	users := []string{}
+	for user := range uniqueUsers {
+		users = append(users, user)
+	}
+	return users
+}
+
+// GetUser returns nil if the user is invalid, Login name otherwise
+func GetUser(user *github.User) *string {
+	if user == nil {
+		return nil
+	}
+	return user.Login
+}
+
+// GetUsersList returns the list of valid users
+func GetUsersList(users []*github.User) []string {
+	allUsers := []string{}
+
+	for _, user := range users {
+		validUser := GetUser(user)
+		if validUser != nil {
+			allUsers = append(allUsers, *validUser)
+		}
+	}
+	return mergeUsers(allUsers)
+}
+
+// CreateActiveUsersFromIssue creates a new ActiveUsers object from an issue's fields
+func CreateActiveUsersFromIssue(issue *github.Issue) *ActiveUsers {
+	return createActiveUsers(
+		issue.Assignee,
+		issue.Assignees,
+		issue.User,
+	)
+}
+
+// GetIssueAuthor returns the author of the issue
+func GetIssueAuthor(issue *github.Issue) string {
+	return CreateActiveUsersFromIssue(issue).Author()
+}
+
+// GetIssueAssignees returns the list of assignees
+func GetIssueAssignees(issue *github.Issue) []string {
+	return CreateActiveUsersFromIssue(issue).Assignees()
+}
+
+// GetIssueUsers returns all users of an issue
+func GetIssueUsers(issue *github.Issue) []string {
+	return CreateActiveUsersFromIssue(issue).AllUsers()
+}
+
+func createActiveUsers(assignee *github.User, assignees []*github.User, author *github.User) *ActiveUsers {
+	authorName := GetUser(author)
+	if authorName == nil {
+		authorName = new(string)
+	}
+
+	return &ActiveUsers{
+		assignees: mergeUsers(
+			GetUsersList(assignees),
+			GetUsersList([]*github.User{assignee})),
+		author: *authorName,
+	}
+}
+
+// MentionUsers adds a @ before each name in the list
+func MentionUsers(users []string) []string {
+	mentionedUsers := []string{}
+
+	for _, user := range users {
+		mentionedUsers = append(mentionedUsers, "@"+user)
+	}
+
+	return mentionedUsers
+}
+
+// JoinMentions is a utility function to add @ sign and seperate with space
+func JoinMentions(users []string) string {
+	return strings.Join(MentionUsers(users), " ")
+}
+
+// Assignees returns the list of assignees
+func (u *ActiveUsers) Assignees() []string {
+	return u.assignees
+}
+
+// Author returns the Author of the pull-request/issue
+func (u *ActiveUsers) Author() string {
+	return u.author
+}
+
+// AllUsers return a list of unique users (both assignees and author)
+func (u *ActiveUsers) AllUsers() []string {
+	return mergeUsers(u.Assignees(), []string{u.Author()})
+}
+
+// IsAssignee checks if a user is an assignee
+func (u *ActiveUsers) IsAssignee(user *github.User) bool {
+	if GetUser(user) == nil {
+		return false
+	}
+	username := *GetUser(user)
+
+	for _, assignee := range u.Assignees() {
```

Comment:
Maybe better to just store a sets.String here instead, then you can just do `return u.assignees.Has(username)`.